string type
---
int32 id
